Integrations/PerceptionPointIncidentsHandler.yml

```diff
@@ -0,0 +1,242 @@
+        if url:
+            parsed = urlparse.urlparse(url)
+            next_cursor = urlparse.parse_qs(parsed.query).get('_cursor')
+            return int(next_cursor[0])
+        return -1
+
+
+    def build_url_request_params(last_cursor=None):
+        ARGUMENTS_LIST = [{'demisto_param': 'fetch_malicious',
+                           'req_pname': 'verbose_verdict',
+                           'req_pval': MALICIOUS},
+                          {'demisto_param': 'fetch_blocked',
+                           'req_pname': 'verbose_verdict',
+                           'req_pval': BLOCKED},
+                          {'demisto_param': 'fetch_spam',
+                           'req_pname': 'verbose_verdict',
+                           'req_pval': SPAM}]
+        params_dict = defaultdict(list)
+        arguments_input = demisto.params()
+        arguments_input.update(demisto.args())
+        for darg in ARGUMENTS_LIST:
+            darg_input = arguments_input.get(darg['demisto_param'])
+            if darg_input:
+                params_dict[darg['req_pname']].append(darg.get('req_pval', darg_input))
+        for reqname, reqval in params_dict.iteritems():
+            params_dict[reqname] = ','.join(reqval)
+        if last_cursor:
+            params_dict['_cursor'] = last_cursor
+        return params_dict
+
+
+    def create_incident(record):
+        record.pop('Attachment', None)
+        record['RawJSON'] = json.dumps(record)
+        return record
+
+
+    def collect_incidents(last_cursor=None):
+        list_url = build_request_url(LIST)
+        api_res = get_pp_api_result(url=list_url, last_cursor=last_cursor)
+        num_of_results = api_res['count']
+        incidents = []
+        api_loops = 0
+        while num_of_results and api_loops < API_MAX_LOOPS:
+            incidents += map(create_incident, api_res['results'])
+            if api_res['next']:
+                api_res = get_pp_api_result(url=api_res['next'])
+                num_of_results = api_res['count']
+                api_loops += 1
+        return incidents
+
+
+    def report_incidents(incidents_list):
+        demisto.incidents(incidents_list)
+
+
+    def get_pp_api_result(url, last_cursor=None):
+        res = requests.get(url=url,
+                           params=build_url_request_params(last_cursor),
+                           headers=HEADER)
+        res.raise_for_status()
+        try:
+            res_content = res.json()
+        except:
+            res_content = {}
+        return res_content
+
+
+    def build_request_url(api_action):
+        return URL.format(endpoint=API_ACTIONS_DICT[api_action])
+
+
+    def command_fetch_incidents():
+        try:
+            last_run_id = int(demisto.getLastRun().get('scan_id', -1))
+            incidents_list = collect_incidents(last_run_id)
+            report_incidents(incidents_list)
+            if incidents_list:
+                last_run_id = max(last_run_id, int(incidents_list[-1]['Scan Id']))
```

Comment:
We usually avoid accessing directly to a dictionary value like this: `dict['key']` instead we prefer using the `get` operator like this: dict.get(`key`) to avoid unnecessary errors.

Reply:
fixed